### ISSUE TITLE
Precompile LINQPad queries

### ISSUE THREAD

**Issue by leidegre** [tessin/cloud-pad#19]
A LINQPad script a bit of XML with an optional connection string and assembly references.

When you run a LINQPad script, LINQPad does a lot of _stuff_ to make sure your compiled code acts like a script. For example, it starts an AppDomain and uses remoting to talk to it. All the compiled code gets loaded into this thing and is the run there.

The thing is, you can sort of use reflection to discover the compiled stuff from within the script. What you find is a directory in the %TEMP% folder with a lot of stuff. Like this:

```
handle
query_gxpsoe.dll
query_gxpsoe.pdb
query_hoqumq.dll
query_hoqumq.pdb
query_ndfvxd.dll
query_ndfvxd.pdb
query_tmofxq.dll
query_tmofxq.pdb
query_zbxuhc.dll
query_zbxuhc.pdb
TypedDataContext_ppzhxg.dll
yinmcsgu.autosave
```

Each change is it's own version of the `query_` assemblies and your compiled data context is found in the `TypedDataContext_` assembly.

To run this query you basically new up the `UserQuery` class in the `query_` assembly with a IDbConnection if used with a data context. You then call `Main` to execute the script.

So, I started looking into ways to do this and how LINQPad sets things up. And while it depends on a ton of ambient context it doesn't seem to break if you run things outside a LINQPad context. That is, unless you have a script which depends on some really special LINQPad service, you should be able to new up and run these `UserQuery` classes yourself.

There will be some constraints and what might be doable within the script itself, but I cannot see how this couldn't work at least in the common case.

The obvious benefit of precompiled LINQPad queries is that they don't take 10 seconds to start up. And we don't have to maintain a connection to a separate process.

If we did this, I think we should do it as a complement to the existing way of deploying LINQPad scripts in Azure. I don't know if we should bother with AppDomain isolation but it might be difficult to managed different script dependencies without it. i.e. it's it would be bad if two seperate LINQPad scripts had to agree on the same version of a specific assembly and maybe break stuff if they didn't. Also, use shadow copying stuff to ease deployment/redeployment.

NuGet packages are actually trivial to deal with because of this little trick.

```csharp
foreach (var loadedAssembly in AppDomain.CurrentDomain.GetAssemblies())
{
	if (loadedAssembly.IsDynamic)
	{
		loadedAssembly.FullName.Dump("Dynamic");
		continue;
	}
	loadedAssembly.Location.Dump();
}
```

Which results in stuff like this:

```
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_tfadgm.dll
C:\Windows\Microsoft.NET\Framework\v4.0.30319\mscorlib.dll
C:\Program Files (x86)\LINQPad5\LINQPad.exe
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll
!Location LINQPad.Common, Version=1.0.0.0, Culture=neutral, PublicKeyToken=21353812cd2a2db5 
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll
!Location Resources, Version=1.0.0.0, Culture=neutral, PublicKeyToken=21353812cd2a2db5 
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml.Linq\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.Linq.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Data.Linq\v4.0_4.0.0.0__b77a5c561934e089\System.Data.Linq.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_32\System.Data\v4.0_4.0.0.0__b77a5c561934e089\System.Data.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\Microsoft.CSharp\v4.0_4.0.0.0__b03f5f7f11d50a3a\Microsoft.CSharp.dll
!Location Hyperlinq, Version=1.0.0.0, Culture=neutral, PublicKeyToken=21353812cd2a2db5 
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Security\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Security.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_ndfvxd.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\TypedDataContext_ppzhxg.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_32\System.Transactions\v4.0_4.0.0.0__b77a5c561934e089\System.Transactions.dll
C:\NuGet.FW46\FW46\Newtonsoft.Json 9.0.1\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_32\System.EnterpriseServices\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.EnterpriseServices.dll
Dynamic Anonymously Hosted DynamicMethods Assembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null 
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Caching\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Runtime.Caching.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Numerics\v4.0_4.0.0.0__b77a5c561934e089\System.Numerics.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Serialization\v4.0_4.0.0.0__b77a5c561934e089\System.Runtime.Serialization.dll
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\Accessibility\v4.0_4.0.0.0__b03f5f7f11d50a3a\Accessibility.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_tmofxq.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_hoqumq.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_zbxuhc.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_gxpsoe.dll
C:\Users\leidegre\AppData\Local\Temp\LINQPad5\_tbhfjbag\query_tfadgm.dll
```

As we can see here assemblies without a `Location`  are loaded from memory by LINQPad and everything else if dependencies that we should bundle with the exception of framework stuff. I made a trivial console application to test this out and it works.

**Comment by leidegre** [comment]
https://docs.microsoft.com/en-us/dotnet/framework/app-domains/shadow-copy-assemblies

**Comment by leidegre** [comment]
Fixed via 2.0 release https://github.com/tessin/cloud-pad/releases/tag/2.0.0
